### DISCUSSION TITLE
Can we put invoking callback in the services map machine's config? Always produce wrong event type with Typegen.

### DISCUSSION THREAD

**Discussion by azamuddin** [statelyai/xstate#3860] [Q&A]
Can we put invoking callback in the services map machine's config? In typescript, it always produce wrong type when using inline vs serializing it?

I'm using Typegen to generate typings. (`xstate typegen src/ --watch`)

Consider this: 

I have a machine that invoke callback on one of it's state: 

```ts
invoke: {
  id: 'documentListener', 
  src: (ctx, ev) => (send, onEvent) => {
     // Do something 
     // and use `send`
     send({type: 'HELLO'});
  }, 
  on: {
       HELLO: { // .... do something }
   }
}
```

It's all good. But when I move the callback implementation to service like this: 

```ts
invoke: {
  id: 'documentListener', 
  src: `documentListener`,
  on: {
       HELLO: { // .... do something }
   }
}
```

Then the machine config:

```
{
  services: {
     documentListener: (ctx, ev) => (send, onEvent) => {
          // NOW suddenly this give type error
          send({type: 'HELLO'});
      }
  }
}
```

The error tells me that "HELLO" event is not supported, but only the internal platform supported such as: 

- `""` empty type 
- "done.invoke.documentMoveListener"
- "error.invoke.documentMoveListener"

I'm on: 
- `xstate` version `4.36.0` 

Is there anything I missed here?

**Comment by davidkpiano** [comment]
Can you share the full code?

Also, this looks wrong:

```diff
invoke: {
  id: 'documentListener', 
  src: `documentListener`,
- on: {
-      HELLO: { // .... do something }
- }
}
+ on: {
+      HELLO: { // .... do something }
+ }
```

**Reply by azamuddin** [reply]
Yes, the on property should be on the outside of invoke declaration. That's what I'm doing on actual code. 

Okay, I here's the funny thing. 

I tried to reproduce minimal code using xstate react template on Codesandbox, but guess what? It works perfectly in there. 

But on my nvim & vscode, it give type error. My typescript version is 4.9.5.

**Comment by azamuddin** [comment]
@davidkpiano okay, so if I remove the "tsTypes" from machine, it will work fine. But if use it, the type will error. 

Check this codesandbox (line 100 to test) https://codesandbox.io/s/relaxed-pike-e7dm5y?file=/src/index.ts
